src/overview/actions.js

```diff
@@ -257,12 +269,70 @@ export const showTags = index => (dispatch, getState) => {
 }
 
 export const filterTag = tag => (dispatch, getState) => {
-    const query = selectors.query(getState())
+    const state = getState()
+    const query = selectors.query(state)
+    const filterTags = selectors.filterTags(state)
+
     const transformedTag = `#${tag.split(' ').join('+')} `
 
-    const newQuery = query.includes(transformedTag)
+    const newQuery = query.includes(transformedTag.slice(0, -1))
         ? query.replace(transformedTag, '') // Either remove it, if already there
         : transformedTag + query // or prepend it, if not there
 
-    dispatch(setQuery(newQuery))
+    dispatch(setQueryTagsDomains(newQuery))
+}
+
+export const setQueryTagsDomains = (input, isEnter) => (dispatch, getState) => {
+    const state = getState()
+    const filterTags = selectors.filterTags(state)
+    const filterDomains = selectors.filterDomains(state)
+    let filterStatus = selectors.showFilter(state)
+    const query = input
+
+    const terms = input.toLowerCase().match(/\S+/g) || []
+
+    terms.forEach(term => {
+        if (input[input.length - 1] === ' ' || isEnter) {
+            if (constants.HASH_TAG_PATTERN.test(term)) {
+                const tag = term
+                    .slice(1)
+                    .split('+')
+                    .join(' ')
+
+                input = input.replace(isEnter ? term : term + ' ', '')
+
+                if (filterTags.indexOf(tag) === -1) {
+                    dispatch(addTagFilter(tag))
+                } else {
+                    dispatch(delTagFilter(tag))
+                }
+
+                if (!filterStatus) {
+                    filterStatus = !filterStatus
+                    dispatch(showFilter())
+                }
+            }
+
+            if (constants.DOMAIN_TLD_PATTERN.test(term)) {
+                input = input.replace(isEnter ? term : term + ' ', '')
+
+                if (filterDomains.indexOf(term) === -1) {
+                    dispatch(addDomainFilter(term))
+                } else {
+                    dispatch(delDomainFilter(term))
+                }
+
+                if (!filterStatus) {
+                    filterStatus = !filterStatus
+                    dispatch(showFilter())
+                }
+            }
+        }
+    })
+
+    dispatch(setQuery(input))
+
+    if (!query.includes('#')) {
```

Comment:
This really shouldn't be here. It's overriding the search debounce (managed in `overview/epics` module) by manually calling search every input change (apart from when '#' is present). The `setQuery` is already dispatched for each `change` event, so if you delete this part and just add `actions.setQuery.getType()` back into the epics module, it should work a lot nicer. Let me know if there's a reason that I'm overlooking.

Reply:
@poltak I removed from the `epic` because we have to stop the reload the result when we type for the tag, so that when we type `#` then we have to stop the reload the result till the tag goes into the tag filter.

Reply:
Why would you need to do that? The query builder will extract the tags found in the input state and still work fine. Like if I wanted to type "github #javascript" and I start typing it fast, but I stop for at least 500ms (the debounce time) on "github #javas", it will still search fine for the term "github" and use the tag "#javas". Now it will stop searching if there is a "#" in the input state.

Reply:
This is the expected behaviour by @oliversauter, when we type # then we have to stop the search result updating.

Reply:
Then the main thing would be trying to make that behaviour work without breaking the debounce. The debounce is managed by the epic so it would need to go back in, but maybe you could put this `.includes('#')` check at the start of the `search` thunk to `return` early. That way, the debounce would still work and searches will be sent off, but they'll just do nothing if there is a '#' in state.